scene/resources/surface_tool.cpp

```diff
@@ -285,16 +297,15 @@ Array SurfaceTool::commit_to_arrays() {
 				array.resize(varr_len);
 				PoolVector<Vector3>::Write w = array.write();
 
-				int idx = 0;
-				for (List<Vertex>::Element *E = vertex_array.front(); E; E = E->next(), idx++) {
-					const Vertex &v = E->get();
+				for (unsigned int n = 0; n < vertex_array.size(); n++) {
```

Comment:
```suggestion
				for (uint32_t n = 0; n < vertex_array.size(); n++) {
```

Bit of a nitpick, but you use ``unsigned int`` in some places and ``uint32_t``. For consistency its probably better to just use one unless I am missing something

Reply:
Afaik it's fairly innocuous which to choose for a counter, so consistency is as good a reason as other and I agree. I'm guilty of often choosing by the surrounding code, or how the moon is shining that particular night... but I _usually_ go for `uint32_t`. :grin:

Extra obligatory waffle: I'm no expert on them but array counters do seem a hotbed of debate amongst nerds, and have sparked a few threads on twitter. I've heard some suggest that `unsigned int` is better for counter as theoretically the native size _could_ be quicker on a platform, but most sources seem to suggest a 64 bit processor should be equal speed for 32 or 64 bit int (and on 32 bit processor, `unsigned int` would be the same as `uint32_t`). What might be more relevant is what you are comparing it _against_ (though optimizer will often sort this I expect).

The real religious arguments appear with signed / unsigned counters / `size()` functions, with some citing integer overflow as being undefined behaviour, but unsigned being more tricky than signed for decrementing loops. Although afaik all the modern CPUs use two's complement, so in practice signed overflow tends to have the same behaviour. And you would seldom design a loop with an overflow.

I also know one guy who insisted on reverse iterating loops because checking against zero is faster. But then you'd have to load cache lines in reverse so no idea if that would work.